tripal/tripal
# Publishing content types and fields linked to a property type_id

# Tripal 4 Core Dev Task

### Issue #1515


### Tripal Version:  4.x

## Description


This PR allows you to publish content types and their fields when the content type is tied to a `type_id` in a "prop" table rather than a `type_id` in the base table.  Different types of analyses are the primary example of this. 

## Testing?



1. Create one more more Analysis content types
2. Create a Genome Assembly content type
3. Delete all of these new entities.
4. Publish the Analysis content type. This should result in restoration of all of the analyses you created before and the ndeleted.
5. Publish the Genome Assembly type. You should only have one type left.
6. Since we don't have functionl tests for some of the issues that were discovered when testing PR #1729 It would be good to test a gene with contacts, and two different properties with multiple values each.  The delete it and publish that to make sure all still works.
